code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -15,13 +15,20 @@
 	var/wall_mounted = 0 //never solid (You can always pass over it)
 	var/health = 100
 	var/lastbang
+	var/can_weld_shut = 1
 	var/max_mob_size = MOB_SIZE_HUMAN //Biggest mob_size accepted by the container
 	var/mob_storage_capacity = 3 // how many human sized mob/living can fit together inside a closet.
-	var/storage_capacity = 30 //This is so that someone can't pack hundreds of items in a locker/crate
-							  //then open it in a populated area to crash clients.
+	var/storage_capacity = 30 //This is so that someone can't pack hundreds of items in a locker/crate then open it in a populated area to crash clients.
+	var/cutting_tool = /obj/item/weapon/weldingtool
+	var/open_sound = 'sound/machines/click.ogg'
+	var/cutting_sound = 'sound/items/Welder.ogg'
+	var/material_drop = /obj/item/stack/sheet/metal
 
 /obj/structure/closet/New()
 	..()
+	if(type == /obj/structure/closet && ticker && ticker.current_state == GAME_STATE_PREGAME && prob(20)) //only roundstart generic closets

```

Comment:
That seems to be a really shitty thing for anyone who maps. As you can no longer properly map a crate that will actually be a crate for sure.

Would be a lot better with some new item that has chance to be a crate or cardboard box, and then editing that to the map. Instead of forcing 20% of crates that are and ever will be to be a cardboarbox.